File: ts-sdk/src/sdk/models/shared/bankingcircleconfig.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class BankingCircleConfig extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=authorizationEndpoint" })
  authorizationEndpoint: string;

  @SpeakeasyMetadata({ data: "json, name=endpoint" })
  endpoint: string;

  @SpeakeasyMetadata({ data: "json, name=password" })
  password: string;

  @SpeakeasyMetadata({ data: "json, name=username" })
  username: string;
}

File: python-sdk/src/formanceapi/models/shared/bankingcircleconfig.py
import dataclasses
from dataclasses_json import dataclass_json
from formanceapi import utils


@dataclass_json
@dataclasses.dataclass
class BankingCircleConfig:
    authorization_endpoint: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('authorizationEndpoint') }})
    endpoint: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('endpoint') }})
    password: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('password') }})
    username: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('username') }})
    